### PR TITLE
Add Onefuzz.container.reset to the SDK

### PR DESCRIPTION
**Pull request by anshuman-goel**
## Summary of the Pull Request

To reduce overload for scaling and for testing I need to delete particular set of container. Adding `container reset` cli argument.

## PR Checklist
* [ ] Applies to work item: #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/onefuzz) and sign the CLI.
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

Also, 
- changing `reset --containers` to call `containers reset`
- sorting types in `ContainerType`

## Validation Steps Performed

Manually tested.

### CODE REVIEW

File: src/cli/onefuzz/api.py [hunk 1]
```diff
@@ -274,6 +274,61 @@ def list(self) -> List[responses.ContainerInfoBase]:
         self.logger.debug("list containers")
         return self._req_model_list("GET", responses.ContainerInfoBase)
 
+    def reset(
```
Comment (bmc-msft) [inline src/cli/onefuzz/api.py:277]: This should use:

`container_type: Optional[List[ContainerType]] = None`

Comment (bmc-msft) [reply src/cli/onefuzz/api.py:277]: Can you add a helper method to ContainerType that indicates the types that are reset by default?

File: src/cli/onefuzz/api.py [hunk 2]
```diff
@@ -274,6 +274,28 @@ def list(self) -> List[responses.ContainerInfoBase]:
...
+        self, container_types: Optional[List[enums.ContainerType]] = None
+    ) -> None:
+        """Reset all Conatiners unless specified.
+        [Caution]: It can lead to unexpected results.
+        """
+        if not container_types:
+            container_types = enums.ContainerType.__dict__["_member_map_"].values()
```
Comment (bmc-msft) [inline src/cli/onefuzz/api.py:284]: It's easier to just iterate over the enum directoy.  `[x for x in enums.ContainerType]`

However, this erases *all* container types, including those we probably don't want to erase, like the various tools containers (such as `afl-linux`)

File: src/cli/onefuzz/api.py (hunk 1 above)
Comment (anshuman-goel) [reply src/cli/onefuzz/api.py:277]: updated and used same default as defined in earlier in `reset --container`

File: src/cli/onefuzz/api.py [hunk 3]
```diff
@@ -274,6 +274,29 @@ def list(self) -> List[responses.ContainerInfoBase]:
         self.logger.debug("list containers")
         return self._req_model_list("GET", responses.ContainerInfoBase)
 
+    def reset(
+        self, container_types: Optional[List[enums.ContainerType]] = None
+    ) -> None:
+        """Reset containers unless specified.
+        [Caution]: It can lead to unexpected results.
```
Comment (anshuman-goel) [inline src/cli/onefuzz/api.py:281]: @bmc-msft The cli is not able to read the lines after first one. Some bug exists over there.

File: src/cli/onefuzz/api.py (hunk 2 above)
Comment (anshuman-goel) [reply src/cli/onefuzz/api.py]: using `reset_defaults` with suggested iterator.

**Review by bmc-msft — APPROVED** [review]

**Pull request merged**